lib/slayer/form.rb

```diff
@@ -1,17 +1,31 @@
+require 'virtus'
+require 'dry-validation'
+
 module Slayer
   class Form
-    extend ModelAttribute
-    include ActiveModel::Validations if defined?(Rails)
+    attr_reader :errors
+    include Virtus.model
 
-    # Internal: Form Class Methods
     class << self
-      def from_params(params)
-        raise NotImplementedError
-      end
+      attr_accessor :_schema
 
-      def from_model(model)
-        raise NotImplementedError
+      def validations(&block)
+        self._schema = Dry::Validation.Schema(&block)
       end
     end
+
+    def validate!
+      result  = self.class._schema.call(self.attributes)
+      @errors = result.errors
+    end
+
+    def valid?
+      validate!
+      @errors.nil? || @errors.empty?
+    end
+
+    def invalid?
```

Comment:
It feels a little strange to be using dry-validations, but to be explicitly going against their opinions on how validations should work (see: http://solnic.eu/2015/12/28/invalid-object-is-an-anti-pattern.html linked from: http://dry-rb.org/gems/dry-validation/comparison-with-activemodel/).

That being said, I think it makes sense to maintain the "Rails API", so that we can in the future support configuring which system to use and so that this can slot into existing Rails projects with less fuss.

So, overall, I think this is the right decision, but something I wanted to call out as...mildly annoying.